### DISCUSSION TITLE
How many models can I run at once?

### DISCUSSION THREAD

**Discussion by mikemcm22** [firemodels/fds#13723] [General]
Hello, my company creates models in Pyrosim and then exports FDS models to run on some servers we have (FDS local). Most of the servers have 20 cores/ 20 logical processors. Some have more but I want to create models based on the most limiting since I am unsure which will be available when I am ready to run. Right now, I have been creating models with 9 meshes. I was told by the previous fire modeler, who has since left the company, that a general rule of thumb is to to make the MPI threads (XX) equal to the number of meshes and the OpenMP threads (YY) equal to whatever number multiplied by XX is < # cores (and not # logical processors). So here 9 x 2 = 18 < 20. Is this strategy correct? Can I run more than one model at the same time on a single server without creating a longer run time? If so would I make OpenMp threads = 1 or would I keep it the same? 

Based on my server limitations, what is the most effective way to get faster run times? Open to other suggestions. Right now I use that general rule of thumb above and make sure that each mesh is under 250k cells typically. 

I have a large project with multiple buildings where I am suppose to run 12 models for 4-5 different buildings and I am trying to get them done as efficiently as I can. I have only created a few so far so changing the number of meshes is also a possibility. But assuming I stick to 9 meshes for now since the buildings are all similar, can I run 2 at a time on each of the 3 servers I have?

**Comment by drjfloyd** [comment]
How many core you can run FDS on at once before you get a negative benefit is going to be hardware and OS dependent. The User's Guide talks about how to do scaling tests and those inputs are in the FDS repository. 

In general you will see the most benefit using your cores for MPI processes rather than OpenMP threads.

**Reply by mikemcm22** [reply]
Does my general rule of thumb sound correct? I was told that I could potentially run two of my 9 mesh models on one 20 core server, I was not sure if I should approach that with one or two  OpenMPI each. Or would you need more information to determine. I know I need to dive into the user guide and do some testing but given my timeline I am trying to see if I can make any quick changes to improve efficiency now.

**Reply by rmcdermo** [reply]
You should use MPI processes instead of OMP threads unless the number of cells in the meshes becomes less than about 5000 cells (roughly 16^3).  At that point, if you have double the number of cores left over, then you can bump OMP threads to 2.  And so on.

**Reply by mikemcm22** [reply]
Okay, so would that mean on a 20 core machine I can run two 9 mesh models at the same time, and specify OMP to be 1 for each if I understand correctly?

**Reply by drjfloyd** [reply]
With 20 cores you can run two 9 mesh models with 9 MPI processses and no OMP threads and not have the models using the same cores.

**Reply by rmcdermo** [reply]
Correct.  Then you would only have 2 cores remaining.  If you had a 36 core machine and your meshes were already small (hence no improved efficiency from more MPI), then you could up OMP to 2, which would parallelize each mesh block.

**Reply by mikemcm22** [reply]
Awesome thank you both
